src/hooks/__tests__/use-track-component.test.tsx

```diff
@@ -0,0 +1,23 @@
+import {renderHook, act} from '@testing-library/react-hooks'
+import {useTrackComponent} from '../use-track-component'
+import {track} from 'utils/analytics'
+
+jest.mock('utils/analytics')
+
+beforeEach(() => {
+  jest.clearAllMocks()
+})
+
+test('it calls track when it first renders', () => {
+  renderHook(useTrackComponent)
```

Comment:
The way this hook is supposed to work is that when the parent component first renders, it will trigger a `track` call. This `renderHook` call will trigger the initial `useEffect` call.